extensions/shared/external-media/sources/with-media.js

```diff
@@ -53,6 +54,45 @@ export default function withMedia() {
 				};
 			}
 
+			componentDidMount() {
+				this.removeArrowKeysPropagationHandler = this.attachArrowKeysPropagationHandler();
+			}
+
+			componentWillUnmount() {
+				if ( this.removeArrowKeysPropagationHandler ) {
+					this.removeArrowKeysPropagationHandler();
+				}
+			}
+
+			attachArrowKeysPropagationHandler = () => {
+				const eventListenerEl = document.querySelector( '.components-modal__content' );
+				if ( ! eventListenerEl ) {
+					return null;
+				}
+
+				eventListenerEl.addEventListener( 'keydown', this.stopArrowKeysPropagation );
+				return () => {
+					eventListenerEl.removeEventListener( 'keydown', this.stopArrowKeysPropagation );
+				};
+			};
+
+			stopArrowKeysPropagation = event => {
+				/**
+				 * When the External Media modal is open, pressing any arrow key causes
+				 * it to close immediately. This is happening because the keydown event
+				 * propagates outside the modal, triggering a re-render and a blur event
+				 * eventually. We could avoid that by isolating the modal from the Image
+				 * block render scope, but it is not possible in current implementation.
+				 *
+				 * This handler makes sure that the keydown event doesn't propagate further,
+				 * which fixes the issue described above while still keeping arrow keys
+				 * functional inside the modal.
+				 */
```

Comment:
Now that we've isolated this to a core modal issue, maybe we could mention the ongoing issue here? https://github.com/WordPress/gutenberg/issues/22940

Basically, how do we know to remove this code once it's fixed in the core Modal?

Reply:
Added in https://github.com/Automattic/jetpack/pull/16055/commits/7337df3b19d2b86dd9d0bb235f38a8886544299f